pkg/utils/cmd/tiller-hack/tiller-hack.go

```diff
@@ -0,0 +1,182 @@
+
+var magicGzip = []byte{0x1f, 0x8b, 0x08}
+
+// BuildClient returns kubernetes clientset
+func BuildClient(kubeconfig string) (*kubernetes.Clientset, error) {
+	var config *rest.Config
+	var err error
+
+	kubeconfigEnv := os.Getenv("KUBECONFIG")
+
+	if kubeconfigEnv != "" {
+		kubeconfig = kubeconfigEnv
+	}
+
+	if kubeconfig != "" {
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	} else {
+		config, err = rest.InClusterConfig()
+	}
+	if err != nil {
+		return nil, err
+	}
+	k8s, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return k8s, nil
+}
+
+func getConfigmapData(k8sClient *kubernetes.Clientset, configMapName, configMapNamespace string) ([]byte, error) {
+
+	cfgMap, err := k8sClient.Core().ConfigMaps(configMapNamespace).Get(configMapName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("fail to get configmap %s/%s with error: %+v", configMapNamespace, configMapName, err)
+	}
+	data, ok := cfgMap.Data["release"]
+	if !ok {
+		return nil, fmt.Errorf("fail to locate \"release\" key in configmap %s/%s", configMapNamespace, configMapName)
+	}
+	return []byte(data), nil
+}
+
+func decodeRelease(data []byte) (*rspb.Release, error) {
+	// base64 decode string
+	b, err := b64.DecodeString(string(data))
+	if err != nil {
+		return nil, err
+	}
+
+	if bytes.Equal(b[0:3], magicGzip) {
+		r, err := gzip.NewReader(bytes.NewReader(b))
+		if err != nil {
+			return nil, err
+		}
+		b2, err := ioutil.ReadAll(r)
+		if err != nil {
+			return nil, err
+		}
+		b = b2
+	}
+
+	var rls rspb.Release
+	// unmarshal protobuf bytes
+	if err := proto.Unmarshal(b, &rls); err != nil {
+		return nil, err
+	}
+	return &rls, nil
+}
+
+func encodeRelease(rls *rspb.Release) (string, error) {
+	b, err := proto.Marshal(rls)
+	if err != nil {
+		return "", err
+	}
+	var buf bytes.Buffer
+	w, err := gzip.NewWriterLevel(&buf, gzip.BestCompression)
+	if err != nil {
+		return "", err
+	}
+	if _, err = w.Write(b); err != nil {
+		return "", err
+	}
+	w.Close()
+
+	return b64.EncodeToString(buf.Bytes()), nil
+}
+
+func updateConfigMap(k8sClient *kubernetes.Clientset, configMapName, configMapNamespace, updatedData string) error {
+	cfgMap, err := k8sClient.Core().ConfigMaps(configMapNamespace).Get(configMapName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("fail to get configmap %s/%s with error: %+v", configMapNamespace, configMapName, err)
+	}
+	cfgMap.Data["release"] = updatedData
+
+	err = retry.RetryOnConflict(retry.DefaultBackoff, func() (err error) {
+		_, err = k8sClient.CoreV1().ConfigMaps(configMapNamespace).Update(cfgMap)
+		return
+	})
+	return nil
+}
+
+func main() {
+	flag.Parse()
+	flag.Set("logtostderr", "true")
+
+	k8sClient, err := BuildClient(*kubeconfig)
+	if err != nil {
+		logrus.Errorf("Failed to create a client: %+v\n", err)
+		os.Exit(1)
+	}
+	if *configMapName == "" {
+		logrus.Errorf("configmap name is missing.")
+		os.Exit(1)
+	}
+	if *configMapNamespace == "" {
```

Comment:
why not have a default of `kube-system` - I don't even know if it is possible to run tiller in any other namespace but `kube-system`...

Reply:
It is, it's quite a common RBAC model for tiller on multi-tenant clusters.
Not sure how applicable to the istio use case it is though

Reply:
I agree to keep namespace for flexibility, what I can do is to make namespace to default to `kube-system` namespace if somebody ever need to override it, the parameter will be there.

Reply:
fyi, see the magic-namespace chart. we use it in multitenant k8s's.